### PR TITLE
6603771: Nimbus L&F: Ctrl+F7 keybinding for Jinternal Frame throws a NPE.

### PR DESCRIPTION
**Pull request by prsadhuk**
In Swingset2 JInternalFrame demo, focusing on any internalframes and pressing Ctrl+F7 and then any of up/down/left/arrow key to move the internal frames results in NPE in NimbusL&F.
It is because `JComponent.processKeyBinding() `calls `SwingUtilities.notifyAction` which calls `BasicDesktopPaneUI.actionPerformed` where it gets `Desktop.minOnScreenInsets` property to get the Insets and then uses insets.top/bottom/left/right value to move the internalframe but this property is not defined for Nimbus so deferencing Insets object results in NPE.

Fix is added to create a 0 insets incase some L&F do not define` Desktop.minOnScreenInsets` property.
No regression test is added as it can be easily checked by SwingSet2 JInternalFrame demo.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-6603771](https://bugs.openjdk.org/browse/JDK-6603771): Nimbus L&F: Ctrl+F7 keybinding for Jinternal Frame throws a NPE.

### Reviewers
 * [Abhishek Kumar](https://openjdk.org/census#abhiscxk) (@kumarabhi006 - Author) ⚠️ Review applies to [051b325c](https://git.openjdk.org/jdk/pull/11605/files/051b325c1a5495852b53d8b7236b7e5baa68af7d)
 * [Sergey Bylokhov](https://openjdk.org/census#serb) (@mrserb - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11605/head:pull/11605` \
`$ git checkout pull/11605`

Update a local copy of the PR: \
`$ git checkout pull/11605` \
`$ git pull https://git.openjdk.org/jdk pull/11605/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11605`

View PR using the GUI difftool: \
`$ git pr show -t 11605`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11605.diff">https://git.openjdk.org/jdk/pull/11605.diff</a>

</details>

### CODE REVIEW

**Review by kumarabhi006 — APPROVED** [review]
Verified in SwingSet2 demo.

**Comment by mrserb** [comment]
It would be good to add a test case to make sure this bug will not happen someday again.

**Comment by prsadhuk** [comment]
> It would be good to add a test case to make sure this bug will not happen someday again.

I tried but it seems AWT EventQueue is gobbling up the NPE and not throwing it to application which is why even though NPE is seen but test does not fail...

```
/*
 * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 *
 * This code is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License version 2 only, as
 * published by the Free Software Foundation.
 *
 * This code is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
 * version 2 for more details (a copy is included in the LICENSE file that
 * accompanied this code).
 *
 * You should have received a copy of the GNU General Public License version
 * 2 along with this work; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
 * or visit www.oracle.com if you need additional information or have any
 * questions.
 */
/*
 * @test
 * @key headful
 * @bug 6603771
 * @summary  Verifies Nimbus L&F: Ctrl+F7 keybinding for Jinternal Frame throws a NPE
 * @run main JInternalFrameTest
 */
import java.awt.*;
import java.awt.event.*;
import javax.swing.*;

public class JInternalFrameTest {
    static JFrame jFrame;
    static JInternalFrame iFrame;

     private static void setLookAndFeel(UIManager.LookAndFeelInfo laf) {
        try {
            UIManager.setLookAndFeel(laf.getClassName());
//	    UIManager.setLookAndFeel("javax.swing.plaf.nimbus.NimbusLookAndFeel");
        } catch (UnsupportedLookAndFeelException ignored) {
            System.out.println("Unsupported L&F: " + laf.getClassName());
        } catch (ClassNotFoundException | InstantiationException
                | IllegalAccessException e) {
            throw new RuntimeException(e);
        }
    }

    public static void main(String[] args) throws Exception {
        Robot robot = new Robot();
        for (UIManager.LookAndFeelInfo laf :
                 UIManager.getInstalledLookAndFeels()) {
            System.out.println("Testing L&F: " + laf.getClassName());
            SwingUtilities.invokeAndWait(() -> setLookAndFeel(laf));
            try {
                SwingUtilities.invokeAndWait(() -> createUI());
                robot.waitForIdle();
                robot.delay(1000);

                Point pt = iFrame.getLocationOnScreen();
                Rectangle dim = iFrame.getBounds();
                robot.mouseMove(pt.x + dim.width/3, pt.y);
                robot.waitForIdle();
                robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
                robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
                robot.waitForIdle();
                robot.delay(1000);
                robot.keyPress(KeyEvent.VK_CONTROL);
                robot.keyPress(KeyEvent.VK_F7);
                robot.keyRelease(KeyEvent.VK_F7);
                robot.keyRelease(KeyEvent.VK_CONTROL);
                robot.waitForIdle();
                robot.delay(1000);
                robot.keyPress(KeyEvent.VK_UP);
                robot.keyRelease(KeyEvent.VK_UP);
                robot.waitForIdle();
                robot.delay(1000);
            } catch (Exception e) {
                System.out.println("NPE");
		e.printStackTrace();
            }finally {
                if (jFrame != null) {
                    SwingUtilities.invokeAndWait(() -> jFrame.dispose());
                }
            }
        }
    }
   
    private static void createUI() {	
     	jFrame = new JFrame();

	JDesktopPane desktopPane = new JDesktopPane();

        iFrame =  new JInternalFrame("Test");

	iFrame.setTitle("InternalFrame");
	iFrame.setClosable(true);
	iFrame.setMaximizable(true);
	iFrame.setIconifiable(true);
	iFrame.setResizable(true);
	iFrame.setLocation(50, 50);
        iFrame.setSize(200, 200);
        iFrame.setVisible(true);

	desktopPane.add(iFrame);

	JPanel panel = new JPanel();
	panel.setLayout(new BorderLayout());
	panel.add(desktopPane, BorderLayout.CENTER);

	jFrame.add(panel, BorderLayout.CENTER);
        jFrame.setSize(400, 400);
	jFrame.setLocationRelativeTo(null);
        jFrame.setVisible(true);
	
    }
}    
```

**Comment by mrserb** [comment]
You can install the `UncaughtExceptionHandler` on EDT to catch that exception:
```java
    public static void main(String[] args) throws Exception {
        EventQueue.invokeAndWait(
                () -> Thread.currentThread().setUncaughtExceptionHandler(
                        (t, e) -> System.out.println("Error?")));
```

**Comment by mrserb** [comment]
Looks like there are many "Whitespace error" reported by jcheck in the test.

File: test/jdk/javax/swing/JInternalFrame/JInternalFrameTest.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+                robot.delay(1000);
+                robot.keyPress(KeyEvent.VK_UP);
+                robot.keyRelease(KeyEvent.VK_UP);
+                robot.waitForIdle();
+                robot.delay(1000);
+                if (failed) {
+                    throw new RuntimeException("Exception thrown");
+                }
+            } finally {
+                if (jFrame != null) {
```
Comment (mrserb) [inline test/jdk/javax/swing/JInternalFrame/JInternalFrameTest.java:104]: If `jFrame` is checked on the main thread it should be volatile as well.

**Review by mrserb — APPROVED** [review]

File: test/jdk/javax/swing/JInternalFrame/JInternalFrameTest.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+            }
+        }
+    }
+
+    private static void createUI() {
+        jFrame = new JFrame();
+
+        JDesktopPane desktopPane = new JDesktopPane();
+
+        iFrame =  new JInternalFrame("Test");
```
Comment (turbanoff) [inline test/jdk/javax/swing/JInternalFrame/JInternalFrameTest.java:118]:
```suggestion
        iFrame = new JInternalFrame("Test");
```

**Pull request closed without merging**